File: 02-04-2026/cap_fibonacci.js
function capitalizeFibonacci(str) {
  const fibSequence = [0, 1, 1, 2, 3, 5, 8, 13, 21, 34];

  if (str.length === 0) return str;
  else if (fibSequence.at(-1) >= str.length)
    return capitalizeFibonacciIndex(str, fibSequence);
  else {
    const fibSeq = generateFibonacciSequence(str);
    return capitalizeFibonacciIndex(str, fibSeq);
  }
}

const generateFibonacciSequence = (str, sequence = [0, 1]) => {
  while (sequence[sequence.length - 1] < str.length) {
    const nextFib =
      sequence[sequence.length - 1] + sequence[sequence.length - 2];
    sequence.push(nextFib);
  }
  return sequence;
};

const capitalizeFibonacciIndex = (str = "", fibSequence = [0, 1]) => {
  let [strLength, newStr] = [str.length, ""];
  for (let i = 0; i < strLength; i++) {
    if (fibSequence.includes(i)) newStr += str[i].toUpperCase();
    else newStr += str[i].toLowerCase() || str[i];
  }
  return newStr;
};

console.log(capitalizeFibonacci("hello world")); // "HELLo woRld".
console.log(capitalizeFibonacci("HELLO WORLD")); // "HELLo woRld".
console.log(capitalizeFibonacci("hello, world!")); // "HELLo, wOrld!".
console.log(
  capitalizeFibonacci("The quick brown fox jumped over the lazy dog."),
); // "THE qUicK broWn fox jUmped over thE lazy dog.".
console.log(
  capitalizeFibonacci(
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin pulvinar ex nibh, vel ullamcorper ligula egestas quis. Integer tincidunt fringilla accumsan. Integer et metus placerat, gravida felis at, pellentesque nisl.",
  ),
); // "LOREm ipSum dOlor sit amet, consecTetur adipiscing elit. proin pulvinar ex nibh, vel ullaMcorper ligula egestas quis. integer tincidunt fringillA accumsan. integer et metus placerat, gravida felis at, pellentesque nisl.".

File: 02-04-2026/cap_fibonacci.py
def capitalize_fibonacci(s):
    fibSequence = [0, 1, 1, 2, 3, 5, 8, 13, 21, 34]

    if len(s) == 0: return s;
    elif fibSequence[-1] >= len(s):
        return capitalizeFibonacciIndex(s, fibSequence);
    else:
        fibSeq = generateFibonacciSequence(s);
        return capitalizeFibonacciIndex(s, fibSeq);

def generateFibonacciSequence(s, sequence = [0, 1]):
  while sequence[len(sequence) - 1] < len(s):
    nextFib = sequence[len(sequence) - 1] + sequence[len(sequence) - 2];
    sequence.append(nextFib);
  return sequence;

def capitalizeFibonacciIndex(s = "", fibSequence = [0, 1]):
  strLength, newStr = len(s), "";
  for i in range(strLength):
    if i in fibSequence: newStr += s[i].upper();
    else: newStr += s[i].lower() or s[i];
  return newStr;


print(capitalize_fibonacci("hello world")); # "HELLo woRld".
print(capitalize_fibonacci("HELLO WORLD")); # "HELLo woRld".
print(capitalize_fibonacci("hello, world!")); # "HELLo, wOrld!".
print(
  capitalize_fibonacci("The quick brown fox jumped over the lazy dog."),
); # "THE qUicK broWn fox jUmped over thE lazy dog.".
print(
  capitalize_fibonacci(
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin pulvinar ex nibh, vel ullamcorper ligula egestas quis. Integer tincidunt fringilla accumsan. Integer et metus placerat, gravida felis at, pellentesque nisl.",
  ),
); # "LOREm ipSum dOlor sit amet, consecTetur adipiscing elit. proin pulvinar ex nibh, vel ullaMcorper ligula egestas quis. integer tincidunt fringillA accumsan. integer et metus placerat, gravida felis at, pellentesque nisl.".
